core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -184,14 +185,15 @@ Query parsePercolatorDocument(String id, BytesReference source) {
         }
     }
 
+    //norelease this method parses from xcontent to lucene query, need to re-investigate how to split context here
     private Query parseQuery(String type, XContentParser parser) {
```

Comment:
So do you think we need to do the parsing to QueryBuilder already in this PR or should we do it later?